commit fe59d5372ae719ca4550958f1e5bb4dd6eeac9cd
tree 08ac6bf26961f7bfba6c371f566333c030dd5d77
parent bcf5111a58c7db968c3fb9cd77e340a5e076f549
author Russell King <rmk@dyn-67.arm.linux.org.uk> 1159719247 +0100
committer Russell King <rmk+kernel@arm.linux.org.uk> 1159719247 +0100

[SERIAL] Fix resume handling bug

Unfortunately, pcmcia_dev_present() returns false when a device is
suspended, so checking this on resume does not work too well.  Omit
this test.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

backported patch from the fedora patchset. -maks

diff --git a/drivers/serial/serial_cs.c b/drivers/serial/serial_cs.c
index cbf260b..06a246a 100644
--- a/drivers/serial/serial_cs.c
+++ b/drivers/serial/serial_cs.c
@@ -185,14 +185,12 @@ static int serial_suspend(struct pcmcia_
 
 static int serial_resume(struct pcmcia_device *link)
 {
-	if (pcmcia_dev_present(link)) {
-		struct serial_info *info = link->priv;
-		int i;
+	struct serial_info *info = link->priv;
+	int i;
 
-		for (i = 0; i < info->ndev; i++)
-			serial8250_resume_port(info->line[i]);
-		wakeup_card(info);
-	}
+	for (i = 0; i < info->ndev; i++)
+		serial8250_resume_port(info->line[i]);
+	wakeup_card(info);
 
 	return 0;
 }
